### ISSUE TITLE
Icon only buttons are not accessible

### ISSUE THREAD

**Issue by mhemesath** [cerner/terra-core#1058]
# Issue Description
Icon only buttons fail accessibility testing.

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
Passes accessibility tests

## Current Behavior
```
To solve this violation, you need to:
Fix at least one (1) of these issues:
Element has a value attribute and the value attribute is empty
Element has no value attribute or the value attribute is empty
Element does not have inner text that is visible to screen readers
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible
Element's default semantics were not overridden with role="presentation"
Element's default semantics were not overridden with role="none"
And
Fix the following:
Element is in tab order and does not have accessible text
```

## Steps to Reproduce

1. Install [Axe Chrome Extension](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)
2. Navigate to the [icon only button example page](http://engineering.cerner.com/terra-core/#/tests/button-tests/icon-only) and the [button group icon example](http://engineering.cerner.com/terra-core/#/tests/button-group-tests/icon-only-button-group)
3. Run axe by going to developer tools and clicking axe tab.

## Environment

* Component Version:
* Browser Name and Version:
* Operating System and version (desktop or mobile):

## Additional Information
Tests that are failing from this are being bypassed with a rule of

```js
  'button-name': { enabled: false },
```
Searching for this should identify all known failed tests in terra.

**Comment by PaulSolDev** [comment]
JIRA created

**Comment by mhemesath** [comment]
This is also an issue in button groups: http://engineering.cerner.com/terra-core/#/tests/button-group-tests/icon-only-button-group

**Comment by mhemesath** [comment]
This is also an issue in collapsible menu view http://engineering.cerner.com/terra-core/#/tests/collapsible-menu-view-tests
